core/src/test/java/org/elasticsearch/index/query/AbstractQueryTestCase.java

```diff
@@ -400,6 +406,47 @@ protected static XContentBuilder toXContent(QueryBuilder<?> query, XContentType
     }
 
     /**
+     * Randomize the name of a query builder. Note that this can change the meaning of the query builder. Like <code>match</code> can turn
+     * into <code>match_phrase</code>.
+     */
+    private XContentBuilder randomizeName(XContentBuilder builder, QueryBuilder<?> query) throws IOException {
+        String newName = RandomPicks.randomFrom(random(), getSupportedNames(query));
+        if (newName.equals(query.getName())) {
+            return builder;
+        }
+        Tuple<XContentType, Map<String, Object>> t = XContentHelper.convertToMap(builder.bytes(), true);
+        Object queryAsMap = t.v2().remove(query.getName());
+        if (queryAsMap == null) {
+            throw new IllegalArgumentException(
+                    "Test query claimed to have name [" + query.getName() + "] but instead wrote [" + t.v2() + "]");
+        }
+        t.v2().put(newName, queryAsMap);
+        builder = XContentFactory.contentBuilder(t.v1());
+        builder.map(t.v2());
+        logger.debug("Switched from [{}] to [{}]", query.getWriteableName(), newName);
+        return builder;
+    }
+
+    /**
+     * Get a list of all supported names for a query builder class. Uses reflection and naming conventions so it is brittle.
+     */
+    protected Collection<String> getSupportedNames(QueryBuilder<?> query) {
+        try {
+            return ((QueryBuilder.Names) query.getClass().getField("NAMES").get(null)).all();
```

Comment:
Yeah, I'm not sold on this either. Its really just something I hacked together so we didn't have nothing. I agree that there is only the one place where the names are listed and in this case that is funky. It makes the test feel like it proves less stuff.